### ISSUE TITLE
Waiting for the response of the HttpEndpoint activity spend a long time.

### ISSUE THREAD

**Issue by blue-cloud-net** [elsa-workflows/elsa-core#2583]
Hello,

 I have somthing problem on the elsa 2.3.I don't know where I set it wrong,causing the problem.
 Please help me to solve this problem.

1.Waiting for the response of the HttpEndpoint activity spend a long time.A request takes about 4 seconds.

2.My WorkflowDefinition(count:100+) like this:
> HttpEndpoint
> My customize activities(about 5)
> WriteHttpResponse

3.I expect it only spend less than 0.5s.Please tell me how to solve it?

4.A request takes about 4 seconds or More.But I check the workflowInstance logs,every activity spend less than 0.3s.And In the log,I found the BookmarkIndexer spend 0.8s to index 0 bookmark each time.A request will be indexed bookmark a few times.

5.By debug,I found when run the workflow and activity excuted,it will publish the _WorkflowInstanceSaved_. It will be indexing the bookmarks.
  In the `BookmarkIndexer.IndexBookmarksAsync` methed,used most of the thime calling the `CachingWorkflowRegistry.ListActiveInternalAsync` method.
  In the `CachingWorkflowRegistry.ListActiveInternalAsync` method,there are more than 100 published workflows that need to be traversed.
![1639336785(1)](https://user-images.githubusercontent.com/37796118/145726386-1200730b-2037-4a0a-96b1-cfad94bcee1e.jpg)

6.Order question:
I have set without _Save Workflow Instance_ in the activities, why it also saves workflowInstance after activity excuted?
What the bookmarks does?(Index to resume workflow?)

 My English is not good, if you don’t understand, please let me explain.

**Comment by sfmskywalker** [comment]
Hi @blue-cloud-net 

It's hard to say why your workflow performs poorly without profiling, but it was recently discovered that the code you highlighted in number 5 is one of the causes and has been fixed (available in the `master` branch).

With regards to point 6, I tried to reproduce that behavior, but for me the workflow was saved only twice: once when created, and once when finished. I added 3 WriteLine activities, and they all executed without causing the workflow instance to be saved after they executed (except of course when the workflow ended).

**Comment by blue-cloud-net** [comment]
@sfmskywalker 

Thank you for your reply.
I pull the newest source to test. In the `master` branch, the problem of slow workflow execution has been solved.
The number 6,  may be my problem, I will confirm the problem. I guess the logs of other paralle workflow make that it look like saving multiple times. At that time, I judged based on the number of times to the times of the `BookmarkIndexer` log.
